engine/data_record.cpp

```diff
@@ -28,19 +28,25 @@ StringRecord* StringRecord::PersistStringRecord(
                                       type, status, old_version, key, value,
                                       expired_time);
   if (with_buffer) {
```

Comment:
we use buffer to accelerate pmem_memcpy, withou pmem, we can always call memcpy to addr directly without buffer.